Compile common libraries to single `libs` directory (#1425)
cocotb/cocotb: documentation/source/newsfragments/1425.change.rst

> All libraries are compiled on setup to ``/cocotb/libs`` directory.

Reviewer: ```suggestion
All libraries are compiled during installation to the ``cocotb/libs`` directory.
```

> Interface libraries `vpicocotb/vhpicocotb/fli` are compiled with `_simulator_name` postfix.

Reviewer: ```suggestion
The interface libraries ``libcocotbvpi`` and ``libcocotbvhpi`` have been renamed to have a ``_simulator_name`` postfix.
```

> The ``simulator`` module has moved to :mod:`cocotb.simulator`.

Reviewer: ```suggestion
The ``simulator`` module has moved to :mod:`cocotb.simulator`.
The ``LD_LIBRARY_PATH`` environment variable is no longer modified by cocotb, as the libraries now discover each other via ``RPATH`` settings.
```

Reviewer: I can imagine people being broken somehow, and having `LD_LIBRARY_PATH` appear somewhere in the release notes will likely be the red flag they're looking for.

> `simulator` module is now compiled on setup to ``cocotb`` directory and has to be now imported with ``form cocotb import simulator``.

Reviewer: ```suggestion
The ``simulator`` module has moved to ``cocotb.simulator``.
```
